nixops_gcp/gcp_common.py

```diff
@@ -11,75 +11,47 @@
+
 
 def ensure_not_empty(value, name):
     if not value:
         raise Exception("{0} must not be empty".format(name))
 
+
 def ensure_positive(value, name):
     if value <= 0:
         raise Exception("{0} must be a positive integer".format(name))
 
-class ResourceDefinition(nixops.resources.ResourceDefinition):
 
-    def __init__(self, xml):
-        nixops.resources.ResourceDefinition.__init__(self, xml)
-
-        res_name = self.get_option_value(xml, 'name', str)
-        if len(res_name)>63 or re.match('[a-z]([-a-z0-9]{0,61}[a-z0-9])?$', res_name) is None:
-            raise Exception("resource name ‘{0}‘ must be 1-63 characters long and "
-              "match the regular expression [a-z]([-a-z0-9]*[a-z0-9])? which "
-              "means the first character must be a lowercase letter, and all "
-              "following characters must be a dash, lowercase letter, or digit, "
-              "except the last character, which cannot be a dash.".format(res_name))
-
-        self.copy_option(xml, 'project', str)
-        self.copy_option(xml, 'serviceAccount', str)
-        self.access_key_path = self.get_option_value(xml, 'accessKey', str)
-
-    def get_option_value(self, xml, name, type, optional = False,
-                         empty = True, positive = False):
-        types = { str: '/string', int: '/int', bool: '/bool', 'resource': '', 'strlist': '/list' }
-        xml_ = xml.find("attrs")
-        xml__= xml_ if xml_ is not None else xml
-        elem = xml__.find("attr[@name='%s']%s" % (name, types[type]))
-
-        if type == str:
-            value = optional_string(elem)
-        elif type == int:
-            value = optional_int(elem)
-        elif type == bool:
-            value = optional_bool(elem)
-        elif type == 'resource':
-            value = ( optional_string(elem.find("string")) or
-                      self.get_option_value(elem, 'name', str, optional = True) )
-        elif type == 'strlist':
-            value = sorted( [ s.get("value")
-                              for s in elem.findall("string") ] ) if elem is not None else None
-
-        if not optional and value is None:
-            raise Exception("option {0} must be set".format(name))
-
-        if not empty:
-            ensure_not_empty(value, name)
-        if positive:
-            ensure_positive(value, name)
-        return value
-
-    # store the option value in a property, following the naming conventions
-    # by converting "optionName" to "option_name"
-    def copy_option(self, xml, name, type, optional = False,
-                         empty = True, positive = False):
-      setattr(self, re.sub(r'([a-z])([A-Z])',r'\1_\2', name).lower(),
-              self.get_option_value(xml, name, type, optional = optional,
-                                    empty = empty, positive = positive) )
+class ResourceDefinition(nixops.resources.ResourceDefinition):
+    def __init__(self, name, config):
+        super().__init__(name, config)
```

Comment:
The evaluation is failing because of this

```tTraceback (most recent call last):
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-gcp-SjyPoyyC-py3.7/bin/nixops", line 11, in <module>
    load_entry_point('nixops', 'console_scripts', 'nixops')()
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-gcp-SjyPoyyC-py3.7/src/nixops/nixops/__main__.py", line 705, in main
    args.op(args)
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-gcp-SjyPoyyC-py3.7/src/nixops/nixops/script_defs.py", line 316, in op_info
    do_eval(depl)
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-gcp-SjyPoyyC-py3.7/src/nixops/nixops/script_defs.py", line 224, in do_eval
    depl.evaluate()
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-gcp-SjyPoyyC-py3.7/src/nixops/nixops/deployment.py", line 480, in evaluate
    defn = _create_definition(name, cfg, cfg["targetEnv"])
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-gcp-SjyPoyyC-py3.7/src/nixops/nixops/deployment.py", line 1692, in _create_definition
    return cls(name, nixops.resources.ResourceEval(config))
  File "/home/tewfikghariani/repositories/nix-community/nixops-gce_master/nixops_gcp/backends/gce.py", line 44, in __init__
    super().__init__(name, config)
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-gcp-SjyPoyyC-py3.7/src/nixops/nixops/backends/__init__.py", line 52, in __init__
    super().__init__(name, config)
  File "/home/tewfikghariani/repositories/nix-community/nixops-gce_master/nixops_gcp/gcp_common.py", line 37, in __init__
    super().__init__(name, config)
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-gcp-SjyPoyyC-py3.7/src/nixops/nixops/resources/__init__.py", line 45, in __init__
    if not issubclass(config_type, ResourceOptions):
TypeError: issubclass() arg 1 must be a class
```